packages/math/src/decimal.ts

```diff
@@ -118,7 +115,7 @@ export class Decimal {
     }
 
     this.data = {
-      atomics: new BN(atomics),
+      atomics: BigInt(atomics),
       fractionalDigits: fractionalDigits,
     };
   }
```

Comment:
Slightly out of scope, but we can now also change the constuctor argument to `atomics: string | bigint` for more efficientcy in a few places.